pkg/domain/infra/abi/network.go

```diff
@@ -144,6 +172,33 @@ func (ic *ContainerEngine) NetworkExists(ctx context.Context, networkname string
 
 // Network prune removes unused cni networks
 func (ic *ContainerEngine) NetworkPrune(ctx context.Context, options entities.NetworkPruneOptions) ([]*entities.NetworkPruneReport, error) {
+	// get all filters
+	filters, err := netutil.GenerateNetworkPruneFilters(options.Filters)
```

Comment:
Why isn't dangling filters done in the GenerateNetworkPruneFilters function?

Reply:
my understanding is that GenerateNetworkPruneFilters does not know (or has reasons to know) the list of all containers, which is needed by this feature.

Reply:
GenerateNetworkPruneFilters Generates a list of filters These filters can return your danglinFilterFunc?

Reply:
And I could see where I want to Prune all Networks without containers.

Reply:
GenerateNetworkPruneFilters comes from another repository, do you think it would be better to push this change there? However, I am not sure how to let common/libnetwork use the podman's ContainerEngine struct, then.

Reply:
@Luap99 Thoughts?

Reply:
I think it is cleaner to have it in podman. Adding to c/common would require ugly callback function just to get the container networks.